pypsa/networkclustering.py

```diff
@@ -147,12 +147,20 @@ def aggregatebuses(network, busmap, custom_strategies=dict()):
 
 def aggregatelines(network, buses, interlines, line_length_factor=1.0):
 
+    # Make sure network component contents start with "Input" for r and x to avoid missing data in the clustered network before call aggregatelines.
+    network.components["Line"]["attrs"].at['x','status']='Input (required)'
+    network.components["Line"]["attrs"].at['r','status']='Input (required)'
+  
     #make sure all lines have same bus ordering
     positive_order = interlines.bus0_s < interlines.bus1_s
     interlines_p = interlines[positive_order]
     interlines_n = interlines[~ positive_order].rename(columns={"bus0_s":"bus1_s", "bus1_s":"bus0_s"})
     interlines_c = pd.concat((interlines_p,interlines_n), sort=False)
 
+    # line length is an optional input when type is not set so original network may or may not contain this information. If not, line length factor below won't work.
+    if network.lines.length.sum() == 0:
+        interlines_c['length']=haversine_pts(network.buses.loc[interlines_c.bus0, ['x', 'y']], network.buses.loc[interlines_c.bus1, ['x', 'y']])
+    
```

Comment:
I don't think we should have this here. Now we are adding information in the cluster routing which was not there before. I rather suggest we move that feature to `n.calculate_dependent_values()`. What do you think @martacki?